File: frontend/src/api/NamedBusinessObject.js
import BusinessObject from './BusinessObject';

export default class NamedBusinessObject extends BusinessObject {

    constructor(aName) {
        super();
        this.name = aName;
    }

    // setting a new Name for Named BOs
    setName(aName) {
        this.name = aName;
    }

    // getting the Named BOs Name
    getName() {
        return this.name;
    }
}

File: src/server/bo/NamedBusinessObject.py
from src.server.bo import BusinessObject as bo


# Named Business Object Klasse (abstrakt)
class NamedBusinessObject(bo.BusinessObject):

    def __init__(self):
        super().__init__()
        self._name = ""

    def get_name(self):
        return self._name

    def set_name(self, value):
        self._name = value
